### PR TITLE
Clear command parameters and thor -> gli

### PR DESCRIPTION
**Pull request by malakai97**
The first commit simplify the command classes by simply passing the key-value pairs as parameters. This makes things a bit clearer to understand.

The second commit replaces thor with gli. This isn't a huge upgrade, but may be easier to work with. I would be okay not merging this commit.

### CODE REVIEW

File: lib/fauxpaas/cli.rb [hunk 1]
```diff
@@ -1,12 +1,211 @@
...
+
+      desc "Deploy a release"
+      long_desc "Deploys the instance using the source described by the " \
+        "default branch. If a reference is given, that will be deployed " \
+        "instead. The reference can be a branch, tag, or SHA."
+      arg "instance"
+      arg "reference", :optional
+      command :deploy do |c|
+        c.action do |global_options, _options, args|
+          invoker.add(
```
Comment (botimer) [inline lib/fauxpaas/cli.rb]: Is this supposed to be add_command?

File: lib/fauxpaas/cli.rb [hunk 2]
```diff
@@ -1,12 +1,211 @@
...
+      end
+
+      desc "Initiate a rollback"
+      long_desc "Initiate a rollback to the specified cache if given, " \
+        "or the most recent one otherwise. Use with care"
+      arg "instance"
+      arg "cache", :optional
+      command :rollback do |c|
+        c.action do |global_options, _options, args|
+          invoker.add(
```
Comment (botimer) [inline lib/fauxpaas/cli.rb]: Is this supposed to be add_command?

Comment (malakai97) [reply lib/fauxpaas/cli.rb]: fixed

File: lib/fauxpaas/cli.rb (hunk 1 above)
Comment (malakai97) [reply lib/fauxpaas/cli.rb]: fixed

**Pull request merged**